app/aboot/lk2nd-motorola.c

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+#include <debug.h>
+#include <libfdt.h>
+#include <malloc.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "lk2nd-device.h"
+#include "lk2nd-motorola.h"
+#include "smem.h"
+
+#define SMEM_ID_VENDOR0          0x86
+#define SMEM_KERNEL_RESERVE      SMEM_ID_VENDOR0
+#define SMEM_KERNEL_RESERVE_SIZE 1024
+
+static void fdt_getprop_u32(const void *fdt, int offset, const char *name, uint32_t *val) {
+	int len;
+	const uint32_t *prop = fdt_getprop(fdt, offset, name, &len);
+	if (len == sizeof(*val))
+		*val = fdt32_to_cpu(*prop);
+	else
+		dprintf(CRITICAL, "Failed to get prop %s in device tree", name);
+}
+
+void lk2nd_motorola_smem_write_unit_info(const void *fdt, int offset)
+{
+	int chosen_offset, len, ret = 0;
+	const char *baseband;
+	struct mmi_unit_info *unit_info;
+
+	fdt_getprop(fdt, offset, "lk2nd,motorola-unitinfo", &len);
+	if (len == 0) {
+		return;
+	}
```

Comment:
either use brackets for simple if statements or don't, but not both in one method please :)